# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: accessible/ipc/DocAccessibleParent.cpp
# Commit: d1558d374cef
# Full Hash: d1558d374cef9d954a84b7db4646e48a4878f621
# Author: James Teh <jteh@mozilla.com>
# Date: 2023-11-30 04:12:04
# Regressor Bug: 1862802
# File Overlap Count: 1
# Description:
#   Bug 1862802 part 1: Provide an argument to PlatformCaretMoveEvent specifying whether the event was caused by user input. r=eeejay
#   
#   This is needed in order to support find in page on Android using remote caret events instead of virtual cursor change events.
#   
#   Depends on D192641
# ==============================================================================

diff -r d6f61c448b90 -r d1558d374cef accessible/ipc/DocAccessibleParent.cpp
--- a/accessible/ipc/DocAccessibleParent.cpp	Tue Nov 28 21:50:11 2023 +0000
+++ b/accessible/ipc/DocAccessibleParent.cpp	Tue Nov 28 22:03:19 2023 +0000
@@ -439,7 +439,8 @@
 mozilla::ipc::IPCResult DocAccessibleParent::RecvCaretMoveEvent(
     const uint64_t& aID, const LayoutDeviceIntRect& aCaretRect,
     const int32_t& aOffset, const bool& aIsSelectionCollapsed,
-    const bool& aIsAtEndOfLine, const int32_t& aGranularity) {
+    const bool& aIsAtEndOfLine, const int32_t& aGranularity,
+    const bool& aFromUser) {
   ACQUIRE_ANDROID_LOCK
   if (mShutdown) {
     return IPC_OK();
@@ -463,7 +464,7 @@
   }
 
   PlatformCaretMoveEvent(proxy, aOffset, aIsSelectionCollapsed, aGranularity,
-                         aCaretRect);
+                         aCaretRect, aFromUser);
 
   if (!nsCoreUtils::AccEventObserversExist()) {
     return IPC_OK();